File: typescript/src/resources/kimi.ts
/** Kimi chat completions resource. */

import { Transport } from '../runtime/transport';

export type KimiModel =
  | 'kimi-k3'
  | 'kimi-k2.6'
  | 'kimi-k2-thinking-turbo'
  | 'kimi-k2.5'
  | 'kimi-k2-thinking'
  | 'kimi-k2-instruct-0905'
  | 'kimi-k2-0905-preview'
  | 'kimi-k2-turbo-preview'
  | 'kimi-k2-0711-preview'
  | (string & {});

class Completions {
  constructor(private transport: Transport) {}

  async create(opts: {
    model: KimiModel;
    messages: Array<Record<string, unknown>>;
    stream?: false;
    [key: string]: unknown;
  }): Promise<Record<string, unknown>>;
  async create(opts: {
    model: KimiModel;
    messages: Array<Record<string, unknown>>;
    stream: true;
    [key: string]: unknown;
  }): Promise<AsyncGenerator<Record<string, unknown>>>;
  async create(opts: {
    model: KimiModel;
    messages: Array<Record<string, unknown>>;
    stream?: boolean;
    [key: string]: unknown;
  }): Promise<Record<string, unknown> | AsyncGenerator<Record<string, unknown>>> {
    const { model, messages, stream, ...rest } = opts;
    const body: Record<string, unknown> = { model, messages, ...rest };

    if (stream) {
      body.stream = true;
      return this.streamResponse(body);
    }
    return this.transport.request('POST', '/kimi/chat/completions', { json: body });
  }

  private async *streamResponse(body: Record<string, unknown>): AsyncGenerator<Record<string, unknown>> {
    for await (const chunk of this.transport.requestStream('POST', '/kimi/chat/completions', { json: body })) {
      yield JSON.parse(chunk);
    }
  }
}

class ChatNamespace {
  readonly completions: Completions;
  constructor(transport: Transport) {
    this.completions = new Completions(transport);
  }
}

export class Kimi {
  readonly chat: ChatNamespace;

  constructor(transport: Transport) {
    this.chat = new ChatNamespace(transport);
  }
}

File: python/src/acedatacloud/resources/kimi.py
"""Kimi chat completions resource."""

from __future__ import annotations

import json as _json
from collections.abc import AsyncIterator, Iterator
from typing import Any, Literal

KimiModel = Literal["kimi-k3", "kimi-k2.6", "kimi-k2-thinking-turbo", "kimi-k2.5", "kimi-k2-thinking"]


class _Completions:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def create(
        self,
        *,
        model: KimiModel,
        messages: list[dict[str, Any]],
        stream: bool = False,
        **kwargs: Any,
    ) -> dict[str, Any] | Iterator[dict[str, Any]]:
        body = {"model": model, "messages": messages, **kwargs}
        if stream:
            body["stream"] = True
            return self._stream(body)
        return self._transport.request("POST", "/kimi/chat/completions", json=body)

    def _stream(self, body: dict[str, Any]) -> Iterator[dict[str, Any]]:
        for chunk in self._transport.request_stream("POST", "/kimi/chat/completions", json=body):
            yield _json.loads(chunk)


class _AsyncCompletions:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def create(
        self,
        *,
        model: KimiModel,
        messages: list[dict[str, Any]],
        stream: bool = False,
        **kwargs: Any,
    ) -> dict[str, Any] | AsyncIterator[dict[str, Any]]:
        body = {"model": model, "messages": messages, **kwargs}
        if stream:
            body["stream"] = True
            return self._stream(body)
        return await self._transport.request("POST", "/kimi/chat/completions", json=body)

    async def _stream(self, body: dict[str, Any]) -> AsyncIterator[dict[str, Any]]:
        async for chunk in self._transport.request_stream("POST", "/kimi/chat/completions", json=body):
            yield _json.loads(chunk)


class _Chat:
    def __init__(self, transport: Any) -> None:
        self.completions = _Completions(transport)


class _AsyncChat:
    def __init__(self, transport: Any) -> None:
        self.completions = _AsyncCompletions(transport)


class Kimi:
    """Synchronous Kimi client."""

    def __init__(self, transport: Any) -> None:
        self.chat = _Chat(transport)


class AsyncKimi:
    """Async Kimi client."""

    def __init__(self, transport: Any) -> None:
        self.chat = _AsyncChat(transport)
